File: Restaurant/Restaurant.Web/Modules/ServerTypes/Default/ProductColumns.ts
import { CategortFormatter } from "@/Default/Product/CategryFormatter";
import { ImageViewerFormatter } from "@/Default/Product/ImageViewerFormatter";
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ProductRow } from "./ProductRow";

export interface ProductColumns {
    Id: Column<ProductRow>;
    ProductName: Column<ProductRow>;
    Description: Column<ProductRow>;
    Price: Column<ProductRow>;
    Currency: Column<ProductRow>;
    PriceBefor: Column<ProductRow>;
    Photo: Column<ProductRow>;
    CategoryId: Column<ProductRow>;
}

export class ProductColumns extends ColumnsBase<ProductRow> {
    static readonly columnsKey = 'Default.Product';
    static readonly Fields = fieldsProxy<ProductColumns>();
}

[ImageViewerFormatter, CategortFormatter]; // referenced types

File: Restaurant/Restaurant.Web/Modules/Default/Product/ProductColumns.cs
using Restaurant;
using Serenity.ComponentModel;
using System.ComponentModel;

namespace Restaurant.Default.Columns;

[ColumnsScript("Default.Product")]
[BasedOnRow(typeof(ProductRow), CheckNames = true)]
public class ProductColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId")]
    public int Id { get; set; }
    [EditLink]
    public string ProductName { get; set; }
    public string Description { get; set; }
    public decimal Price { get; set; }
    public string Currency { get; set; }
    public decimal PriceBefor { get; set; }
    [ImageViewerFormatter]
    public string Photo { get; set; }
    [CategortFormatter]
    public int CategoryId { get; set; }
}